dali/pipeline/operators/reader/loader/sequence_loader.cc

```diff
@@ -89,16 +61,51 @@ std::vector<SequenceLoader::Stream> SequenceLoader::ParseStreams(string file_roo
   return streams;
 }
 
-std::vector<size_t> SequenceLoader::CalculateStreamSizes(const std::vector<Stream> &streams,
-                                                         size_t sample_lenght) {
+}  // namespace filesystem
+
+std::vector<size_t> detail::CalculateSequencesCounts(const std::vector<filesystem::Stream> &streams,
+                                                     size_t sequence_lenght) {
   std::vector<size_t> result;
   for (const auto &s : streams) {
-    result.push_back(s.second.size() - (sample_lenght - 1));
+    auto sequences_count =
+        s.second.size() >= (sequence_lenght - 1) ? s.second.size() - (sequence_lenght - 1) : 0;
```

Comment:
can you explain why the -1?

Reply:
Reworked it to support step and stride, it should be clearer now when we collect only full sequences.